src/clj/ataru/codes_service/postal_code_client.clj

```diff
@@ -29,31 +29,44 @@
         value (:metadata koodi)]
     {key value}))
 
+(defn- parse-response
+  [response]
+  (->> response
+       :body
+       slurp
+       .getBytes
+       ByteArrayInputStream.
+       xml/parse
+       :content
+       (map parse-element)
+       (map filter-postal-codes)
+       (into {})))
+
+(defn- load-postal-codes
+  []
+  (parse-response @(http/get (str (get-in config [:codes-service :url]) "/rest/posti/koodi"))))
+
+(def ^:private memo-postal-codes
+  (memoize load-postal-codes))
```

Comment:
Note to self: should probably conditionally memoize only when a valid response has been received.